File: 백준/Silver/1497. 기타콘서트/기타콘서트.js
const fs = require('fs');
const input = fs.readFileSync("/dev/stdin").toString().trim().split("\n");

const [N, M] = input[0].split(" ").map(Number);;
const guitars = Array.from({length: N}, () => 0n);

for (let i = 0; i < N; i++) {
    const valiableSongs = input[i+1].split(" ")[1].replaceAll('Y', '1').replaceAll('N', '0');
    guitars[i] = BigInt('0b' + valiableSongs);
}

let maxValue = 0n;
let maxCount = 0;

const backtracking = (currentValue, currentIdx, currentCount) => {
    if (currentValue === maxValue) {
        maxCount = Math.min(maxCount, currentCount);
    }
    if (currentValue > maxValue) {
        maxValue = currentValue;
        maxCount = currentCount;
    }
    
    if (currentIdx === N) return;
    
    backtracking(currentValue | guitars[currentIdx], currentIdx+1, currentCount+1);
    backtracking(currentValue, currentIdx+1, currentCount);
    
}

backtracking(0n, 0, 0);

console.log((maxValue ? maxCount : -1))

File: 백준/Silver/1497. 기타콘서트/기타콘서트.py
import sys
input = sys.stdin.readline

max_value = max_count = 0

def backtracking(current_value, current_idx, current_count):
    global max_value, max_count
    
    if current_value == max_value:
        max_count = min(max_count, current_count)
    if current_value > max_value:
        max_value, max_count = current_value, current_count
        
    if current_idx == N:
        return
        
    backtracking(current_value | guitars[current_idx], current_idx+1, current_count+1)
    backtracking(current_value, current_idx+1, current_count)
    

N, M = map(int, input().split())

guitars = [int(input().split()[1].replace('Y', '1').replace('N', '0'), 2) for _ in range(N)]

backtracking(0, -1, 0)

if max_value == 0:
    print(-1)
else:
    print(max_count)
